src/javascript/app_2/pages/trading/components/elements/language_switcher.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { localize } from '../../../../../_common/localize';
+
+class LanguageSwitcher extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
+            is_collapsed   : false,
+            active_language: this.props.languages[0],
+        };
+    }
+
+    toggleLanguageList = () => {
+        this.setState({
+            is_collapsed: !this.state.is_collapsed,
+        });
+    }
+
+    switchLanguage = (lang) => {
+        this.setState({
+            active_language: lang,
+        });
+    }
+
+    render() {
+        const language_list_collapsed = {
+            visibility: `${this.state.is_collapsed ? 'visible' : 'hidden'}`,
+            position  : 'absolute',
+            width     : '100%',
+        };
```

Comment:
Since we are generally avoiding css-in-js, could you move this to sass instead? :-)

Reply:
@aminroosta , I can move `position` and `width` back to sass but reason why I'm having `visibility` in `style` props is because the `.hidden` class element will flash for an interval until the compiled `css` loads in if it was in sass. 

Hiding it on render looks a lot better. Let me know if there's a better solution for this.

I can't use `display:none` as css3 animations won't trigger if I did.

Reply:
@zakibinary makes sense to me.

Could you add a brief comment on why we have to keep `visibility` in `jsx` and move `position` and `width` to sass.

Reply:
@aminroosta , sure will do.